### ISSUE TITLE
custom_plot_config not changing anything on MultiQC 1.8

### ISSUE THREAD

**Issue by Adrien-Evo** [ewels/MultiQC#1127]
**Description of bug:**
Using the example for custom_plot_config to change the insert size plot from Picard does not result in any changes in the plot. There's no error message either. MultiQC is from conda, so probably not the latest 1.8 version. 

**MultiQC yaml config:**

```yaml
module_order:
    - MACS2_broadPeak_count
    - picard
    - preseq
    - deepTools
    - featureCounts:
        name: 'FRiP'
        info: 'Fraction of reads in peaks (FRiP)'
    - bowtie2:
        name: 'Bowtie2'
        info: 'Bowtie 2 alignment recap' 
    - fastqc:
        name: 'FastQC (trimmed)'
        info: 'This section of the report shows FastQC results after adapter trimming.'
        target: ''
        path_filters:
            - '*val_*fastqc.zip'
    - fastqc:
        name: 'FastQC (raw)'
        path_filters:
            - '*R[0-9]_fastqc.zip'


section_comments:
    read_distribution_profile_plot: 'Accumulated view of the distribution of sequence reads around TSS.'

custom_data:
    encode_metrics:
        plot_type: 'generalstats'
        file_format: 'tsv'
        pconfig:
            - NRF:
                namespace: "NRF"
                title: "NRF"
                description: "ratio between the number of positions in the genome that uniquely mapped reads map to and the total number of uniquely mappable reads"
                max: 1
                min: 0
                scale: 'RdYlGn'
            - PBC1:
                namespace: "PBC1"
                title: "PBC1"
                description: "ratio of genomic locations with EXACTLY one read pair over the genomic locations with AT LEAST one read pair"
                max: 1
                min: 0
                scale: 'RdYlGn'
            - PBC2:
                namespace: "PBC2"
                title: "PBC2"
                description: "ratio of genomic locations with EXACTLY one read pair over the genomic locations with EXACTLY two read pairs"
                max: 10
                min: 0
                scale: 'RdYlGn'

custom_plot_config:
    picard-insertsize:
        cpswitch_c_active: False
        smooth_points: 1000
        xmax: 300

sp:
    encode_metrics:
        fn: '*encodeQC.txt'
```

**MultiQC run details :**

 - MultiQC v1.8
 - Operating System: [eg. macOS Sierra 10.12.5]
 - Python 3.7.3 | packaged by conda-forge

**Comment by ewels** [comment]
Hi @Adrien-Evo,

Sorry to hear you're having problems. I haven't looked into this in detail / tried testing your config yet, but I notice immediately that the example code in https://multiqc.info/docs/#customising-plots looks different:

```yaml
custom_plot_config:
    picard_insert_size:
        xmax: 300
```

Specifically that it's `picard_insert_size` and not `picard-insertsize` as in your config above.

Could this be the problem?

Phil

**Comment by Adrien-Evo** [comment]
Hey thanks for taking a look at it. 
You are right, it's working with : 
```
custom_plot_config:
    picard_insert_size:
        xmax: 300
```

I was actually taking a piece of your multiqc config from [your atacseq nextflow project](https://github.com/nf-core/atacseq/blob/master/assets/multiqc/multiqc_config.yaml)

My bad at not looking back at the doc of multiQC. Thanks !

**Comment by ewels** [comment]
Ah great - thanks for letting me know! @drpatelh - you've got a bug in the atacseq MultiQC config 😱

**Comment by drpatelh** [comment]
Wooooohhhoooooooo! Thanks for spotting @Adrien-Evo and for fixing @ewels :+1:

**Comment by jamigo** [comment]
any idea why this same issue could be happening on Picard's hsmetrics plot?
```
custom_plot_config:
    picard_hsmetrics_target_bases-1:
        xmax: 1000
```
this is a secondary hsmetrics plot we use to separately display mitochondrial and nuclear DNA hsmetrics. I've copy-pasted the plot name from the url in order to make sure that there isn't any typo in it. I've even tested this too, in case the "-1" suffix in the plot name could be the problem, but it was not.
```
custom_plot_config:
    picard_hsmetrics_target_bases:
        xmax: 200
    picard_hsmetrics_target_bases-1:
        xmax: 1000
```
would you prefer me to open a new issue rather than commenting on this one?

**Comment by ewels** [comment]
Always make new issues instead of commenting on closed issues please. It’s very likely that I won’t see activity on closed threads.

But as I’m here - try enclosing the ID in quotes. From GitHub’s YAML code colouring I guess the hyphen is breaking stuff.

**Comment by jamigo** [comment]
you're completely right. sorry for crossposting.

by the way, I found the error: I was using the plot ID from the URL, but the proper ID was not there but in the "Export" section as stated in the documentation. unfortunately the plot I wanted to modify doesn't contain any data beyond xmax=100, so there's really no need to increase the x-axis max value.

**Comment by ewels** [comment]
Yes sorry, there's a bit of a mishmash of IDs used for some config options, it can be kind of confusing.

Maybe check your source code before giving up with the `xmax` thing to be sure that there is no data beyond 100. There's a chance that MultiQC could be cutting that data off - I don't _think_ it should if you change it in the config like that, but worth double checking 👍

**Comment by ewels** [comment]
Just did a quick test and confirmed my suspicion: [this test file](https://github.com/ewels/MultiQC_TestData/blob/master/data/modules/picard/HsMetrics/TestSample.HsMetrics.txt) goes up to 200 but the plot stops at 100X with the following config:

```yaml
custom_plot_config:
    picard_percentage_target_bases:
        xmax: 300
```

![picard_percentage_target_bases](https://user-images.githubusercontent.com/465550/78222709-f79f1a00-74c5-11ea-8a08-873dcbc24a5b.png)

**Comment by ewels** [comment]
Ok, after digging into the code a bit, I figured out what is going on here. The plot is _not_ generated from the `## HISTOGRAM` section of the log file, but rather from the following columns:

* `PCT_TARGET_BASES_1X`
* `PCT_TARGET_BASES_2X`
* `PCT_TARGET_BASES_10X`
* `PCT_TARGET_BASES_20X`
* `PCT_TARGET_BASES_30X`
* `PCT_TARGET_BASES_40X`
* `PCT_TARGET_BASES_50X`
* `PCT_TARGET_BASES_100X`

The code is generic so takes any column with header `PCT_TARGET_BASES_*X`.

I think this is because not all HsMetrics outputs have the `## HISTOGRAM` section (eg. [here](https://github.com/ewels/MultiQC_TestData/blob/master/data/modules/picard/HsMetrics/alignment-hs_metrics.txt)), especially output files with multiple samples (eg. [here](https://github.com/ewels/MultiQC_TestData/blob/master/data/modules/picard/HsMetrics/alignment-hs_metrics_2samps.txt)).

So in summary, I don't think that MultiQC is doing anything wrong here, but this is maybe not super obvious. I'll add something to the Picard module docs for any future googlers.

Phil

**Comment by ewels** [comment]
Of course, we could add code to _also_ plot the Histogram data. I'm not sure how interesting it is though? Quick test for the single example file used in https://github.com/ewels/MultiQC/issues/1127#issuecomment-607676584:

![image](https://user-images.githubusercontent.com/465550/78224068-2ae2a880-74c8-11ea-83e7-719676b8dd4d.png)

**Comment by jamigo** [comment]
wow, this is more than enough. I agree that MultiQC is doing perfectly by plotting what's inside PCT_TARGET_BASES_nX columns, and that if we want to see more coverage information maybe HsMetrics is not the module to aim for.

thanks a lot for all this effort. I know understand much better how MultiQC works, and knowing that x-axis can be so easily modified I'll investigate other tools' output (mosdepth default plots, for instance, already look almost exactly as what I was trying to do), for particular coverage information.

**Comment by ewels** [comment]
Great, sounds good! Thanks for the feedback :+1:
